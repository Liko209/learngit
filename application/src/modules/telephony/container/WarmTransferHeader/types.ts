/*
 * @Author: Shining (shining.miao@ringcentral.com)
 * @Date: 2019-08-28 02:17:01
 * Copyright Â© RingCentral. All rights reserved.
 */
import CallModel from '@/store/models/Call';

type Props = {};

type ViewProps = {
  switchCallItems: CallModel[];
  switchCall: (callId: number) => void;
  endCall: () => void;
  currentCallId: number;
};

export { Props, ViewProps };
