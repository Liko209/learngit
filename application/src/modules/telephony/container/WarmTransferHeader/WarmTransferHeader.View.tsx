/*
 * @Author: Shining (shining.miao@ringcentral.com)
 * @Date: 2019-08-28 02:17:01
 * Copyright Â© RingCentral. All rights reserved.
 */
import * as React from 'react';
import { observer } from 'mobx-react';
import { withTranslation, WithTranslation } from 'react-i18next';
import { ViewProps } from './types';
import { JuiFabButton } from 'jui/components/Buttons';
import { JuiSwitchCallHeader, JuiSwitchCallChip } from 'jui/pattern/Dialer';
import { HOLD_STATE } from 'sdk/module/telephony/entity';

type Props = ViewProps & WithTranslation;

@observer
class WarmTransferHeaderViewComponent extends React.Component<Props> {
  private _EndCall = () => {
    const { t, endCall } = this.props;
    return (
      <JuiFabButton
        color="semantic.negative"
        size="smallMedium"
        showShadow={false}
        tooltipPlacement="bottom"
        iconName="hand_up"
        onClick={endCall}
        tooltipTitle={t('telephony.action.end')}
        aria-label={t('telephony.action.end')}
        data-test-automation-id="cancel-warm-transfer-btn"
      />
    );
  };

  render() {
    const { switchCallItems, switchCall } = this.props;
    return (
      <JuiSwitchCallHeader>
        {switchCallItems.length &&
          switchCallItems.map((item, inx) => {
            const { holdState, displayName, id, time } = item;
            const active = holdState === HOLD_STATE.IDLE;
            return (
              <JuiSwitchCallChip
                active={active}
                name={displayName}
                time={time}
                icon={active ? 'phone' : 'hold'}
                EndCall={inx === 0 ? this._EndCall : undefined}
                onClick={() => switchCall(id)}
                key={holdState}
              />
            );
          })}
      </JuiSwitchCallHeader>
    );
  }
}

const WarmTransferHeaderView = withTranslation('translations')(
  WarmTransferHeaderViewComponent,
);

export { WarmTransferHeaderView };
