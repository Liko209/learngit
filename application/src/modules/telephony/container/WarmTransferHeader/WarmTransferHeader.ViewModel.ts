/*
 * @Author: Shining (shining.miao@ringcentral.com)
 * @Date: 2019-08-28 02:17:01
 * Copyright Â© RingCentral. All rights reserved.
 */

import { computed, observable } from 'mobx';
import { container } from 'framework/ioc';
import { StoreViewModel } from '@/store/ViewModel';
import { Props } from './types';
import { TelephonyStore } from '../../store';
import { TelephonyService } from '../../service';
import { TELEPHONY_SERVICE } from '../../interface/constant';
import { analyticsCollector } from '@/AnalyticsCollector';

class WarmTransferHeaderViewModel extends StoreViewModel<Props> {
  @observable
  private _telephonyService: TelephonyService = container.get(
    TELEPHONY_SERVICE,
  );
  private _telephonyStore: TelephonyStore = container.get(TelephonyStore);

  @computed
  get switchCallItems() {
    return this._telephonyStore.rawCalls.reverse();
  }

  switchCall = (callId: number) => {
    this._telephonyStore.switchCurrentCall(callId);
    this._telephonyService.holdOrUnhold();
  };

  endCall = () => {
    analyticsCollector.cancelTransferCall();
    this._telephonyService.hangUp(this.switchCallItems[0].id);
  };
}

export { WarmTransferHeaderViewModel };
