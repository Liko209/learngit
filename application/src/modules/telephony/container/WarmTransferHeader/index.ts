/*
 * @Author: Shining (shining.miao@ringcentral.com)
 * @Date: 2019-08-28 02:17:01
 * Copyright Â© RingCentral. All rights reserved.
 */

export * from './WarmTransferHeader';
