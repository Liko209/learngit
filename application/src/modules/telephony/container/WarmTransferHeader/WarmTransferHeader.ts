/*
 * @Author: Shining (shining.miao@ringcentral.com)
 * @Date: 2019-08-28 02:17:01
 * Copyright Â© RingCentral. All rights reserved.
 */

import { buildContainer } from '@/base';
import { WarmTransferHeaderView } from './WarmTransferHeader.View';
import { WarmTransferHeaderViewModel } from './WarmTransferHeader.ViewModel';
import { Props } from './types';

const WarmTransferHeader = buildContainer<Props>({
  View: WarmTransferHeaderView,
  ViewModel: WarmTransferHeaderViewModel,
});

export { WarmTransferHeader, Props };
