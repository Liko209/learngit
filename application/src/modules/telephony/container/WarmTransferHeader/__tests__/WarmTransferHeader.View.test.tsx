/*
* @Author: Shining (shining.miao@ringcentral.com)
* @Date: 2019-08-28 02:17:01
* Copyright Â© RingCentral. All rights reserved.
*/
import React from 'react';
import { test, testable } from 'shield';
import { shallow } from 'enzyme';
//import { mountWithTheme } from 'shield/utils';
import { WarmTransferHeaderView } from '../WarmTransferHeader.View';


describe('WarmTransferHeaderView', () => {
  @testable
  class _viewMethods {
    @test('should check text if exist while pass text as props')
    t1() {
      const props = {text: '1234'};
      const wrapper = shallow(<WarmTransferHeaderView {...props} />);
      expect(wrapper.contains(props.text)).toBe(false);
    }
  }
});
