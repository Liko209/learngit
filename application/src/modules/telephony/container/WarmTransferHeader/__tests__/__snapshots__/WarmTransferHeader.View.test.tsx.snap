// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WarmTransferHeaderView _viewMethods should check text if exist while pass text as props 1`] = `<JuiSwitchCallHeader />`;
